packages/core/src/internal/transaction-executor.ts

```diff
@@ -32,20 +32,41 @@ type TransactionWork<T, Tx = Transaction> = (tx: Tx) => T | Promise<T>
 type Resolve<T> = (value: T | PromiseLike<T>) => void
 type Reject = (value: any) => void
 type Timeout = ReturnType<typeof setTimeout>
+type SetTimeout = (callback: (...args: unknown[]) => void, ms: number | undefined, ...args: unknown[]) => Timeout
+type ClearTimeout = (timeoutId: Timeout) => void
+
+interface Dependencies {
+  setTimeout: SetTimeout
+  clearTimeout: ClearTimeout
+}
+
+function setTimeoutWrapper (callback: (...args: unknown[]) => void, ms: number | undefined, ...args: unknown[]): Timeout {
+  return setTimeout(callback, ms, ...args)
+}
+
+function clearTimeoutWrapper (timeoutId: Timeout): void {
+  return clearTimeout(timeoutId)
+}
 
 export class TransactionExecutor {
   private readonly _maxRetryTimeMs: number
   private readonly _initialRetryDelayMs: number
   private readonly _multiplier: number
   private readonly _jitterFactor: number
   private _inFlightTimeoutIds: Timeout[]
+  private readonly _setTimeout: SetTimeout
+  private readonly _clearTimeout: ClearTimeout
   public pipelineBegin: boolean
 
   constructor (
     maxRetryTimeMs?: number | null,
     initialRetryDelayMs?: number,
     multiplier?: number,
-    jitterFactor?: number
+    jitterFactor?: number,
+    dependencies: Dependencies = {
+      setTimeout: setTimeoutWrapper,
```

Comment:
why do you need the wrapper function? you could have `global.setTimeout` it is no longer being manipulated in timers-util